### PR TITLE
Feature refactor trapd hzn 632

### PR DESCRIPTION
**Pull request by pavanrp1**
Added JUNIT modifications for the blueprint. 

Current outstanding issues:
1. Autowiring doesn't seem to be working, looking into it.

### CODE REVIEW

File: features/events/traps/pom.xml [hunk 1]
```diff
@@ -1,126 +1,173 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (soleger) [inline features/events/traps/pom.xml:2]: Please undo these whitespace changes. We use 2-space indentation inside pom.xml files (and most other XML files).

File: features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		this.m_newSuspectOnTrap = m_newSuspectOnTrap;
+	}
+
+	public void setSnmpV3Users(List<SnmpV3User> m_snmpV3Users) {
+		this.m_snmpV3Users = m_snmpV3Users;
+	}
+
+	@Override
+	public String getSnmpTrapAddress() {
+		// TODO Auto-generated method stub
```
Comment (soleger) [inline features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java]: Remove these auto-generated comments.

File: features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	public void setSnmpTrapAddress(String m_snmpTrapAddress) {
+		this.m_snmpTrapAddress = m_snmpTrapAddress;
+	}
+
+	public void setSnmpTrapPort(int m_snmpTrapPort) {
+		this.m_snmpTrapPort = m_snmpTrapPort;
+	}
+
+	public void setM_newSuspectOnTrap(boolean m_newSuspectOnTrap) {
```
Comment (soleger) [inline features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java]: Remove the M_ from the function name and remove the m_ from all of the function arguments here.

File: features/events/traps/src/test/resources/org/opennms/netmgt/trapd/trapdDefault.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (soleger) [inline features/events/traps/src/test/resources/org/opennms/netmgt/trapd/trapdDefault.xml]: Remove this Spring context. You don't need to mock anything for this test besides a TrapdConfigBean and you don't need any of these objects to do that.

If you need a simple TrapQueueProcessorFactory, just make a simple mock object instead. Trying to create a real one using the EventConfDao is too difficult and outside the scope of this unit test.

File: features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdHandlerDefaultIT.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+import org.opennms.netmgt.snmp.TrapProcessor;
+import org.opennms.test.JUnitConfigurationEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.ContextConfiguration;
+
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {
+		"classpath:/META-INF/opennms/emptyContext.xml",
+		//"classpath:/META-INF/opennms/applicationContext-trapDaemon.xml",
```
Comment (soleger) [inline features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdHandlerDefaultIT.java]: Remove everything here except emptyContext.xml.

File: features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapNotificationImpl.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (soleger) [inline features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapNotificationImpl.java]: Delete this class, you don't need to create a new TrapNotification implementation. Please talk to Pradeep about how to create Snmp4JV1TrapInformation or Snmp4JV2TrapInformation objects for testing (which implement TrapNotification).

File: features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapProcessorImpl.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
+/**
```
Comment (soleger) [inline features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapProcessorImpl.java]: You don't need to create a complex TrapProcessor class; you only need to create a mock object for testing. Remove this class and create an inner class inside TrapdHandlerDefaultIT.java instead.

File: features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdHandlerDefaultIT.java [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+		// Leave a bunch of config that won't be available on the Minion side
+		// blank
+		// config.setParser("org.opennms.netmgt.syslogd.CustomSyslogParser");
+		// config.setForwardingRegexp("^.*\\s(19|20)\\d\\d([-/.])(0[1-9]|1[012])\\2(0[1-9]|[12][0-9]|3[01])(\\s+)(\\S+)(\\s)(\\S.+)");
+		// config.setMatchingGroupHost(6);
+		// config.setMatchingGroupMessage(8);
+		// config.setDiscardUei("DISCARD-MATCHING-MESSAGES");
+
+		byte[] messageBytes = "<34>main: 2010-08-19 localhost foo0: load test 0 on tty1\0"
```
Comment (soleger) [inline features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdHandlerDefaultIT.java]: You don't need this message.

File: features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigProcessor.java [hunk 9]
```diff
@@ -28,25 +28,31 @@
...
 
-	public SyslogdConfigProcessor(SyslogdConfig config) {
+	public TrapdConfigProcessor(TrapdConfig config) {
 		m_config = config;
 	}
 
-	public SyslogConnection process(SyslogConnection connection) {
-		connection.setConfig(m_config);
-		return connection;
+	public TrapNotification process(TrapQueueProcessor connection) {
```
Comment (soleger) [inline features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigProcessor.java]: The input and output of this function should be TrapNotification. I'm not sure we're going to need this class so don't make any further changes here.

File: features/events/traps/blueprint-trapd-handler-default.xml [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+		xmlns="http://camel.apache.org/schema/blueprint">
+		<propertyPlaceholder id="properties"
+			location="blueprint:trapHandlerDefaultProperties" />
+
+		<route id="receiveTrapConnection">
+			<from uri="activemq:broadcastTrap" />
+			<convertBodyTo type="org.opennms.netmgt.snmp.TrapNotification" />
+			<!-- <bean ref="unmarshaller"/> -->
+			<!-- Update the TrapdConfig on the message to the local config value -->
+			<bean ref="trapdConfigProcessor" />
```
Comment (soleger) [inline features/events/traps/blueprint-trapd-handler-default.xml]: You can comment out the trapdConfigProcessor endpoint, I don't think we need it here.

**Comment by pavanrp1** [comment]
Done with the changes and pushed the code. Will test this tomorrow and update the status.

Thanks.

**Comment by soleger** [comment]
The whitespace changes are not done yet and the Spring context and activemq-data temp directories need to be removed.

**Comment by soleger** [comment]
I've merged these changes into feature-refactor-trapd. Closing PR.

**Pull request closed without merging**